File: frontend-web/src/types/event.ts
export type Priority = 'Low' | 'Medium' | 'High';
export type RepeatType = 'None' | 'Daily' | 'Weekly' | 'Monthly';

export interface EventRequest {
    title: string;
    description?: string;
    startDate: string; // ISO 8601 string
    endDate: string; // ISO 8601 string
    category: string;
    priority: Priority;
    repeatType: RepeatType;
}

export interface EventResponse {
    id: string;
    title: string;
    description?: string;
    startDate: string;
    endDate: string;
    category: string;
    priority: Priority;
    repeatType: RepeatType;
    creatorId: string;
    creatorName: string;
}

File: backend/src/main/java/com/erp/entity/Event.java
package com.erp.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

/**
 * @file Event.java
 * @description 行事曆事件實體 / Calendar Event Entity
 * @description_en Represents a calendar event with details like category,
 *                 priority, and date range
 * @description_zh 行事曆的事件基本資料，支援優先級、類別與重覆設定
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "events")
@EntityListeners(AuditingEntityListener.class)
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private LocalDateTime startDate;

    @Column(nullable = false)
    private LocalDateTime endDate;

    @Column(nullable = false)
    private String category;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Priority priority;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RepeatType repeatType;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creator_id", nullable = false)
    private User creator;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(nullable = false)
    private LocalDateTime updatedAt;

    public enum Priority {
        Low, Medium, High
    }

    public enum RepeatType {
        None, Daily, Weekly, Monthly
    }
}
